app/controllers/api/v1/requests_controller.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class RequestsController < Api::V1::ApplicationController
+      load_and_authorize_resource :user, find_by: :name
+      before_action :load_request, only: [:confirm, :decline, :cancel]
+      authorize_resource :request, id_param: :request_id, only: [:confirm, :decline, :cancel]
+
+      def index
+        all_requests = @current_user.requests.includes(:debtor, :creditor, :issuer).group_by(&:status)
+
+        mapped_requests = all_requests.transform_values do |requests|
+          requests.map do |t|
+            mapped = {
+              id: t.id,
+              debtor: t.debtor.name,
+              creditor: t.creditor.name,
+              time: t.updated_at,
+              amount: t.amount,
+              message: t.message,
+              issuer: t.issuer.name,
+              actions: []
+            }
+            mapped[:actions] << :confirm if can?(:confirm, t)
+            mapped[:actions] << :cancel if can?(:cancel, t)
+            mapped[:actions] << :decline if can?(:decline, t)
+            mapped
+          end
+        end
+
+        render json: mapped_requests
```

Comment:
Could be a jbuilder so the representation of the request is extracted from the controller code